helm.tf

```diff
@@ -0,0 +1,35 @@
+resource "helm_release" "ingress" {
+  name       = "ingress-nginx"
+  repository = "https://kubernetes.github.io/ingress-nginx"
+  chart      = "ingress-nginx"
+
+  depends_on = [
+    kubernetes_service.test
+  ]
+}
+
+# set kubectl config from terraform output
+resource "null_resource" "after_output" {
+  triggers = {
+    name = "${var.cluster-name}"
+  }
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      mkdir ~/.kube && touch ~/.kube/config
+      terraform output -raw kubeconfig > ~/.kube/config
+    EOT
+  }
+}
+
+# deploy ingress-nginx 
+resource "null_resource" "after_deployment" {
+  triggers = {
+    helm_id = "${helm_release.ingress.id}"
+  }
+
+  provisioner "local-exec" {
+    command = "kubectl apply -f kubernetes-ingress.yaml"
+  }

```

Comment:
This is a good time to introduce you to Destroy-Time Provisioners.  Basically, whenever you're doing a `local-exec` block, Terraform has no way of knowing what the _opposite_ of the command you supplied is.  For instance, if something screwed up and I go and delete just the `null_resource.after_deployment` object from my state, it won't really make any change.

A good practice is to add a destroy-time provisioner (https://www.terraform.io/docs/language/resources/provisioners/syntax.html#destroy-time-provisioners) so that Terraform can handle the reverse of the actual provisioning in case of either Terraform failure or manual deletion of the resource.